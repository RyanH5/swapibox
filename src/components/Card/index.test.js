import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './index';

describe('Card', () => {
  let wrapper;
  let toggleFavorite;
  let id;

  beforeEach(() => {
    wrapper = shallow(<Card
      currentCategory='people'
      name='Jeff'
      homeworld='Florida'
      species='Human'
      population='10000'
      toggleFavorite={jest.fn()}
      id='people Jeff'
      favorites={
        [{
          name: "Luke Skywalker", 
          homeworld: "Tatooine", 
          species: "Human", 
          population: "200000", 
          id: "peopleLuke Skywalker"
        }]
      }
    />);

    toggleFavorite = jest.fn();
    id = "peopleLuke Skywalker";
  });
  it('should match snapshot with all data', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke toggleFavorite when star is clicked', () => {
    wrapper.find('.fav').simulate('click');

    expect(wrapper.find('.active').length.toEqual(1));
  });
});