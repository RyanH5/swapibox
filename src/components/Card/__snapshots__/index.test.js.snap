// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot with all data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentCategory="people"
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "id": "peopleLuke Skywalker",
              "name": "Luke Skywalker",
              "population": "200000",
              "species": "Human",
            },
          ]
    }
    homeworld="Florida"
    id="people Jeff"
    name="Jeff"
    population="10000"
    species="Human"
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="fav inactive"
          onClick={[Function]}
>
          ✰
</span>,
        <h1>
          Name: 
          Jeff
</h1>,
        <h2>
          Homeworld: 
          Florida
</h2>,
        <h3>
          Species: 
          Human
</h3>,
        <h4>
          Homeworld Population: 
          10000
</h4>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "✰",
          "className": "fav inactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "✰",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "Jeff",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "Jeff",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            "Florida",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          "Florida",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            "Human",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          "Human",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld Population: ",
            "10000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld Population: ",
          "10000",
        ],
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="fav inactive"
            onClick={[Function]}
>
            ✰
</span>,
          <h1>
            Name: 
            Jeff
</h1>,
          <h2>
            Homeworld: 
            Florida
</h2>,
          <h3>
            Species: 
            Human
</h3>,
          <h4>
            Homeworld Population: 
            10000
</h4>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "✰",
            "className": "fav inactive",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "✰",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "Jeff",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "Jeff",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              "Florida",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            "Florida",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              "Human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            "Human",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld Population: ",
              "10000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld Population: ",
            "10000",
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
